### PR TITLE
514

### PR DESCRIPTION
**Pull request by looooo**
continued from https://github.com/conda-forge/pyside2-feedstock/pull/59

Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jschueller** [comment]
what does the qt pinning achive here ? is it for osx ?

**Comment by hmaarrfk** [comment]
I think the 5.12.5 was built was on a personal computer for windows and isn't fully isolated. 

I had to do that to get opencv to compile

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-    # https://codereview.qt-project.org/c/pyside/pyside-tools/+/270008
-    - 0001-Make-pyside2-uic-executable-by-windows-shell.patch
     # osx: Symbol not found: __cg_jpeg_resync_to_restart
     # win: ImportError: DLL load failed while importing QtCore: The specified module could not be found.
     - no_pyi.patch  # [osx or win]
 
 build:
-  number: 6
+  number: 0
   skip: true  # [win and vc<14]
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
  skip: true  # [win and vc<14]
  skip: true  # [not linux]
  skip: true  # [py!=37]
```

You should really block other builds from starting. You will get throttled eventually if you try to make quick runs.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by looooo** [comment]
thanks, that was fast ; )

**Comment by hmaarrfk** [comment]
I think the throttling is probably overstated, but at least it will help you focus ;)

**Comment by looooo** [comment]
`'xcb/xcb.h' file not found`

but should be installed in
`host_env/include/xcb/xcb.h`

**Comment by hmaarrfk** [comment]
They seem to spew out hte include directories, maybe try to build locally as it is linux and a little easier to do.

**Comment by hmaarrfk** [comment]
I nuked all the `-dev` packages on my local machine, and tried to do things with:
https://docs.conda.io/projects/conda-build/en/latest/user-guide/recipes/debugging.html

You can try that :/

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
   patches:
-    # https://codereview.qt-project.org/c/pyside/pyside-tools/+/270008
-    - 0001-Make-pyside2-uic-executable-by-windows-shell.patch
     # osx: Symbol not found: __cg_jpeg_resync_to_restart
     # win: ImportError: DLL load failed while importing QtCore: The specified module could not be found.
     - no_pyi.patch  # [osx or win]
 
 build:
-  number: 6
+  number: 0
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
  number: 0
  # https://github.com/conda-forge/qt-feedstock/blob/master/recipe/meta.yaml#L71
  merge_build_host: true  # [linux]
```

Comment (hmaarrfk) [reply recipe/meta.yaml]: Maybe try this.

**Comment by hmaarrfk** [comment]
It seems that it is complaining that:
```
    clang:                              9.0.1-default_hde54327_0   conda-forge
```
and 
```
    libclang:         8.0.1-hc9558a2_1         conda-forge
```
Are incompatible. which now that i look at them closely, obviously. 

That might be a source of the problem.

**Comment by hmaarrfk** [comment]
Can you give me access to your repo?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -54,14 +56,15 @@ requirements:
     - {{ cdt('libxrandr') }}             # [linux]
     - {{ cdt('alsa-lib') }}              # [linux]
     - {{ cdt('xorg-x11-server-xvfb') }}  # [linux]
-
   host:
     - python
-    - qt
+    - qt  5.12.1
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
    - qt 
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -54,14 +56,15 @@ requirements:
...
     - libxml2
     - libxslt
+    - libxcb                             # [linux]
   run:
     - python
+    - qt  5.12
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
    - qt
```

File: recipe/meta.yaml (hunk 3 above)
Comment (hmaarrfk) [reply recipe/meta.yaml]: At least for linux, this allows my environment to converge.

**Comment by hmaarrfk** [comment]
Locally (no docker) i get a little further:
```
/home/mark/miniconda3/conda-bld/pyside2_1578173959058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/include/qt/QtCore/qglobal.h:45:12: fatal: 'type_traits' file not found
 
```

**Comment by looooo** [comment]
@hmaarrfk  feel free to push to my branch. I added you as a collaborator.

**Comment by looooo** [comment]
for linux I guess we have to add `${PREFIX}/include to the include_dirs. But I am not sure how to do this I tried with:
`CXXFLAGS="${CXXFLAGS} -I\ ${PREFIX}/include"`
and
`-DCMAKE_CXX_FLAGS=-I\ ${PREFIX}/include \`
but both ways didn't add the path to the used compile flags.

**Comment by hmaarrfk** [comment]
thats probably because qt rips them out. RIP

**Comment by hmaarrfk** [comment]
Maybe move everything to isystem?

**Comment by hmaarrfk** [comment]
except, they already are in isystem....
```
+CXXFLAGS=-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/pyside2-5.14.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix
```

**Comment by hmaarrfk** [comment]
well somehow it gets further on the CI than on my local machine. who knows. but now it complains about 
```
ImportError: /home/conda/feedstock_root/build_artifacts/pyside2_1578175493213/work/sources/pyside2/build/PySide2/
QtCore.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZN18QMessageLogContext4copyERKS_
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
well i just added a bunch of CDTs and yum_requirements, lets see what happens. likely needs 20 mins.

**Comment by hmaarrfk** [comment]
@jschueller can you speak as to why we have to skip the pyi?

**Comment by hmaarrfk** [comment]
We are back to 
```
qx11info_x11.h:46:10: fatal error: 'xcb/xcb.h' file not found
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
It seems that it is likely due to 
https://github.com/conda-forge/qt-feedstock/blob/master/recipe/build.sh#L83

**Comment by hmaarrfk** [comment]
Even macos fails with the same issue:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=107924&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=113b264d-9551-5065-da10-7eb3bdeaf1bc&l=3402

**Comment by jschueller** [comment]
I think its wiser to go back to 5.12.x

**Pull request closed without merging**